### PR TITLE
Stop skipping matching of request bodies

### PR DESCRIPTION
**Pull request by jackfranklin**
- if a stub doesn't specify `data`, but the request has it, just match as true anyway (so you don't have to define data of every POST request)
- remove the skipping of data matching if the stub is being validated

@harrison can you please review?

### CODE REVIEW

File: spec/stubby.spec.js [hunk 1]
```diff
@@ -298,10 +298,22 @@ describe('stubbing a POST url', function() {
...
+      expect(JSON.parse(xhr.responseText)).toEqual({ a: 1 });
+      done();
+    });
+  });
+
   it('can differentiate between POST and PUT data', function(done) {
     stubby.stub({
       url: '/foobar',
-      method: 'GET'
+      method: 'POST'
```
Comment (jackfranklin) [inline spec/stubby.spec.js]: unrelated change, just a typo I spotted

**Comment by harrison** [comment]
:+1:
